tests/selenium/test_empty.py

```diff
@@ -0,0 +1,45 @@
+import dash
+from dash.dependencies import Input, Output
+from dash.exceptions import PreventUpdate
+
+from dash_table import DataTable
+from dash_html_components import Button, Div
+
+
+def get_app():
+    app = dash.Dash(__name__)
+
+    columns = [{"name": i, "id": i} for i in ["a", "b"]]
+    data = [dict(a=1, b=2), dict(a=11, b=22)]
+
+    app.layout = Div(
+        [
+            Button(id="clear-table", children=["Clear table"]),
+            DataTable(id="table", columns=columns, data=data),
+        ]
+    )
+
+    @app.callback(
+        [Output("table", "data"), Output("table", "columns")],
+        [Input("clear-table", "n_clicks")],
+    )
+    def clear_table(n_clicks):
+        if n_clicks is None:
+            raise PreventUpdate
+
+        nonlocal columns, data
+
+        return (data, columns) if n_clicks % 2 == 0 else (None, None)
+
+    return app
+
+
+def test_empt001_clear_(test):
+    test.start_server(get_app())
+
+    target = test.table("table")
+
+    assert target.is_ready()
+    test.driver.find_element_by_css_selector("#clear-table").click()
+    assert target.is_ready()
+    assert len(test.driver.find_elements_by_css_selector("tr")) == 0
```

Comment:
Once the callback returns, the table should (1) load successfully and get to the ready state, (2) should contain no rows.

TDD'ed locally, can undo redo the fix in the PR if we'd rather have the demonstration.

Reply:
I'm fine with not seeing the test fail on CI - I assume we're all running our own tests with the fix disabled. But it would be nice to have this test check for the proper number of `<tr>` elements before the `clear-table`.

Also, given that you have the `is_ready()` method this may not be helpful, but in wildcards I added a [`wait_for_no_elements` method](https://github.com/plotly/dash/blob/3ec55b5a97fdd34ed8384085df884d86730b1a9c/dash/testing/browser.py#L223)